src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py

```diff
@@ -27,12 +28,66 @@
 @pytest.mark.sdk_test
 @pytest.mark.e2etest
 class TestFlowAsFunc:
-    def test_flow_as_a_func(self):
-        f = load_flow(f"{FLOWS_DIR}/print_env_var")
+    @pytest.mark.parametrize(
+        "test_folder",
+        [
+            f"{FLOWS_DIR}/print_env_var",
+            f"{FLOWS_DIR}/print_env_var_async",
+        ],
+    )
+    def test_flow_as_a_func(self, test_folder):
+        f = load_flow(test_folder)
         result = f(key="unknown")
         assert result["output"] is None
         assert "line_number" not in result
 
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "async_call_folder",
+        [
+            f"{FLOWS_DIR}/print_env_var",
+            f"{FLOWS_DIR}/print_env_var_async",
+        ],
+    )
+    async def test_flow_as_a_func_asynckw(self, async_call_folder):
+        f = load_flow(async_call_folder, is_async_call=True)
+        result = await f(key="PATH")
+        assert result["output"] is not None
+
+    @pytest.mark.asyncio
+    async def test_flow_as_a_func_real_async(self):
+        from promptflow._sdk.entities._flow import AsyncProtectedFlow
+
+        original_async_func = AsyncProtectedFlow.invoke_async
+
+        # Modify the original function and retrieve the time info.
+        run_info = None
+        node_run_infos = None
+
+        async def parse_invoke_async(*args, **kwargs):
+            nonlocal run_info, node_run_infos
+            obj = await original_async_func(*args, **kwargs)
+            run_info = obj.run_info
+            node_run_infos = obj.node_run_infos
+            return obj
+
+        with mock.patch("promptflow._sdk.entities._flow.AsyncProtectedFlow.invoke_async", parse_invoke_async):
+            f = load_flow(f"{FLOWS_DIR}/async_tools", is_async_call=True)
+            result = await f(input_str="Hello")
```

Comment:
maybe we could have several tasks to ensure they are running in parallel

Reply:
Good catch, let me make it here.